pkg/registry/garden/backupinfrastructure/strategy.go

```diff
@@ -76,7 +76,9 @@ func mustIncreaseGeneration(oldBackupInfrastructure, newBackupInfrastructure *ga
 		return true
 	}
 
-	if kutil.HasMetaDataAnnotation(&newBackupInfrastructure.ObjectMeta, common.BackupInfrastructureOperation, common.BackupInfrastructureReconcile) {
+	oldPresent, _ := strconv.ParseBool(oldBackupInfrastructure.ObjectMeta.Annotations[common.BackupInfrastructureForceDeletion])
```

Comment:
Same - error check needed?

Reply:
i'll apply whatever is your opinion over https://github.com/gardener/gardener/pull/1058#discussion_r289813426.